pkg/config/config.go

```diff
@@ -0,0 +1,150 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"os"
+	"strings"
+	"sync"
+)
+
+// constants
+const config_name = "visualization-api"
+const config_dir_path = "/etc/platformvisibility/visualization-api"
+
+/*
+	CONFIG NAMES
+	The following logic has 3 options to specify config value
+		1 - conf file
+		2 - env variable
+		3 - command line argument
+	Conf file values are overriden by env variables,
+	Env variables are overriden by command line arguments
+
+	Example of usage:
+		given name "mysql.port":
+			* config parser would split this string by dot and would try
+				to find "port" option in "mysql" section
+			* env variables parser would replace "." to "_" and would
+				capitalize the strigs, then it would try to find the variable
+				named "MYSQL_PORT"
+			* command line parser would replace "." to "-" and look for
+				"--mysql-port" option
+*/
+const log_level_config_name = "log.level"
+const log_file_config_name = "log.path"
+const error_template = `
+	error in getting config value %s
+    use '%s' in [%s] config file section, or
+	%s env variable, or
+	%s command line flag
+`
+
+type VisualizationApiConfig struct {
+	// logging settings
+	LogFilePath  string
+	LogLevel     string
+	ConsoleDebug bool
+}
+
+var (
```

Comment:
I've used it to make sure that configs would be parsed exactly one time and the result would be stored in module global variable.
The reason I am using sync.Once instead of module init() is -> it gives user an opportunity to decide when this config parse process would be executed.